libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -78,3 +245,355 @@ void test_MQTT_Init_Invalid_params( void )
     mqttStatus = MQTT_Init( &context, &transport, &callbacks, NULL );
     TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
 }
+
+/* ========================================================================== */
+
+/**
+ * @brief Test MQTT_Connect, except for receiving the CONNACK.
+ */
+void test_MQTT_Connect_sendConnect( void )
+{
+    MQTTContext_t mqttContext;
+    MQTTConnectInfo_t connectInfo;
+    MQTTPublishInfo_t willInfo;
+    uint32_t timeout = 2;
+    bool sessionPresent;
+    MQTTStatus_t status;
+    MQTTTransportInterface_t transport;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTApplicationCallbacks_t callbacks;
+    MQTTPacketInfo_t incomingPacket;
+    size_t remainingLength, packetSize;
+
+    setupTransportInterface( &transport );
+    setupCallbacks( &callbacks );
+    setupNetworkBuffer( &networkBuffer );
+    transport.send = transportSendFailure;
+
+    memset( ( void * ) &mqttContext, 0x0, sizeof( mqttContext ) );
+    MQTT_Init( &mqttContext, &transport, &callbacks, &networkBuffer );
+
+    /* Check parameters */
+    status = MQTT_Connect( NULL, &connectInfo, NULL, timeout, &sessionPresent );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+    status = MQTT_Connect( &mqttContext, NULL, NULL, timeout, &sessionPresent );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+    status = MQTT_Connect( &mqttContext, &connectInfo, NULL, timeout, NULL );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+
+    /* Empty connect info fails. */
+    MQTT_GetConnectPacketSize_ExpectAnyArgsAndReturn( MQTTBadParameter );
+    memset( ( void * ) &connectInfo, 0x0, sizeof( connectInfo ) );
+    status = MQTT_Connect( &mqttContext, &connectInfo, NULL, timeout, &sessionPresent );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+
+    connectInfo.pClientIdentifier = CLIENT_IDENTIFIER;
+    connectInfo.clientIdentifierLength = sizeof( CLIENT_IDENTIFIER ) - 1;
+
+    MQTT_GetConnectPacketSize_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_SerializeConnect_ExpectAnyArgsAndReturn( MQTTNoMemory );
+    status = MQTT_Connect( &mqttContext, &connectInfo, NULL, timeout, &sessionPresent );
+    TEST_ASSERT_EQUAL_INT( MQTTNoMemory, status );
+
+    MQTT_SerializeConnect_IgnoreAndReturn( MQTTSuccess );
+
+    /* Transport send failed when sending CONNECT. */
```

Comment:
Please move this line 271 below here to give more context. "transport.send = transportSendFailure;"